src/compiler/variable.ts

```diff
@@ -137,24 +134,31 @@ export function compileVariableDeclaration(state: CompilerState, node: ts.Variab
 				state.usesTSLibrary = true;
 				rhsStr = rhsStr;
 				const id = state.getNewId();
-				preStatements.push(`local ${id} = ${rhsStr}[TS.Symbol_iterator](${rhsStr});`);
+				state.pushPrecedingStatements(
+					node,
+					state.indent + `local ${id} = ${rhsStr}[TS.Symbol_iterator](${rhsStr});\n`,
+				);
 				rhsStr = id;
 			}
 		}
 
-		getBindingData(state, names, values, preStatements, postStatements, lhs, rhsStr);
-		preStatements.forEach(statementStr => (result += state.indent + statementStr + "\n"));
-		if (values.length > 0) {
-			if (isExported && decKind === ts.VariableDeclarationKind.Let) {
-				concatNamesAndValues(state, names, values, false, str => (result += str));
+		let exportVars = false;
+		let noLocal = false;
```

Comment:
The values of these two variables are thrown away. Nitpick, but I would personally prefer
```ts
let exportVars: boolean;
let noLocal: boolean;
```
That way, these have to be assigned by the following logic before use.